sys/mips/mips/trap.c

```diff
@@ -1160,8 +1160,13 @@ trap(struct trapframe *trapframe)
 		msg = "USER_CHERI_EXCEPTION";
 		fetch_bad_instr(trapframe);
 		log_c2e_exception(msg, trapframe, type);
-		i = SIGPROT;
-		ucode = cheri_capcause_to_sicode(trapframe->capcause);
+		if (trapframe->capcause == CHERI_EXCCODE_TLBSTORE) {
```

Comment:
This is the full cause register, not the exception code, so CI test runs show that we still get SIGPROT.
```suggestion
		if ((trapframe->capcause & CHERI_CAPCAUSE_EXCCODE_MASK) >> CHERI_CAPCAUSE_EXCCODE_SHIFT == CHERI_EXCCODE_TLBSTORE) {
```
(but formatted nicely)

Reply:
Nice catch.